appshell/appshell_extensions_mac.mm

```diff
@@ -1175,6 +1175,154 @@ int32 SetMenuTitle(CefRefPtr<CefBrowser> browser, ExtensionString command, Exten
+
+    // convert to std::vector<char *> array.
+    std::vector<char*> argv(1 + args.size(), NULL);
+
+    for (size_t i = 0; i < args.size(); ++i)
+        argv[i] = const_cast<char*>(args[i].c_str());
+    
+    // This is a deprecated API. Apple recommends a dedicated helper to
+    // fix this issue. This ideally should be replaced by SMBless API.
+    status = AuthorizationExecuteWithPrivileges(authorizationRef, tool.c_str(),
+                                                kAuthorizationFlagDefaults, &argv[0], &pipe);
+    
+    if(pipe){
+        fclose(pipe);
+    }
+    
+    return status;
+    
+}
+
+int32 InstallCommandLineTools()
+{
+    // Create authorization reference
+    OSStatus authStatus  = 0;
+    OSStatus toolStatus  = 0;
+    int32    errorCode   = NO_ERROR;
+
+
+    std::string destFile   = "/usr/local/bin/Brackets";
+    std::string destFolder = "/usr/local/bin";
+
+    std::string rmTool     = "/bin/rm";
+    std::string rmArgs     = "-f";
+
+    std::string mkDirTool  = "/bin/mkdir";
+    std::string mkDirArgs  = "-p";
+
+    std::string lnTool     = "/bin/ln";
+    std::string lnArgs     = "-s";
+
+
+    AuthorizationRef authorizationRef = NULL;
+    
+    // AuthorizationCreate and pass NULL as the initial
+    // AuthorizationRights set so that the AuthorizationRef gets created
+    // successfully.
+    // http://developer.apple.com/qa/qa2001/qa1172.html
+    
+    authStatus = AuthorizationCreate(NULL, kAuthorizationEmptyEnvironment,
+                                 kAuthorizationFlagDefaults, &authorizationRef);
+    
+    if(authStatus == errAuthorizationSuccess) {
+        
+        try {
+            // Determine the location of the create the launch script.
+            // We have this file, Brackets.sh, present inside resource folder.
+            
+            NSString* bundlePath = [[NSBundle mainBundle] bundlePath];
+            NSString* sourcePath;
+            NSRange range = [bundlePath rangeOfString: @"/Frameworks/"];
+            
+            if (range.location == NSNotFound) {
+                sourcePath = [[NSBundle mainBundle] pathForResource: @"Brackets" ofType: @"sh"];
+            } else {
+                sourcePath = [bundlePath substringToIndex:range.location];
+                sourcePath = [sourcePath stringByAppendingString:@"/Resources/Brackets.sh"];
+            }
+
+            std::string sourceFile = [sourcePath UTF8String];
+
+            std::vector<std::string> args;
+
+            // Now execute all these steps one by one
+            //  1. Remove existing symlink
+            //  2. Check for existence of the directory and create one if required.
+            //  3  Create symlink at /usr/local/bin.
+            
+            // *** Removing the existing symlink ***
+            
+            args.push_back(rmArgs);
+            args.push_back(destFile);
+            
+            toolStatus = _RunToolWithAdminPrivileges(authorizationRef, rmTool, args);
```

Comment:
We probably need to change our shell script to pick the application from where it was run.

Users don't have to run this script to install command line tools as it is just an alias, and this is the exact reason why we are choosing the alias. The alias is pointing to actual script. Considering a very common scenario where most users install this in /Application folder, when an update is available they would just replace the current Brackets.app in the /Application folder and that ways  user does not have to re-run the script. For this common scenario, there is no need to rerun the install script. 

In other cases of using multiple instances of Brackets, that is where it could by handy rerunning the script as that would make sure the alias now points to the correct app(provided we make the above change of launching the app which encloses the script)